Dockerfile

```diff
@@ -1,30 +1,38 @@
+# syntax=docker/dockerfile:1.7-labs
 FROM nvcr.io/nvidia/pytorch:24.07-py3
 
-# git-lfs is needed to interact with the huggingface hub
+# Install git-lfs for Huggingface hub interaction and sudo for system adjustments
 RUN apt-get update \
-    && apt-get install git-lfs \
+    && apt-get install --no-install-recommends -y git-lfs sudo util-linux \
     && rm -rf /var/lib/apt/lists/* \
     && git lfs install
 
+# Add a user for Fast-LLM with sudo privileges for runtime adjustments
 ARG FAST_LLM_USER_ID=1000
+RUN useradd -m -u $FAST_LLM_USER_ID -s /bin/bash fast_llm \
+    && echo 'fast_llm ALL=(ALL) NOPASSWD: ALL' >> /etc/sudoers
 
-RUN useradd -m -u $FAST_LLM_USER_ID -s /bin/bash -d /home/fast_llm fast_llm
 USER fast_llm
 WORKDIR /app
-ENV PYTHONPATH=/app:/app:/app/Megatron-LM
-ENV PATH=$PATH:/app:/home/fast_llm/.local/bin/
 
-COPY --chown=fast_llm setup.py setup.cfg ./
-COPY --chown=fast_llm fast_llm/__init__.py ./fast_llm/
+# Environment settings for Python and PATH
+ENV PYTHONPATH=/app:/app:/app/Megatron-LM \
+    PATH=$PATH:/home/fast_llm/.local/bin/
 
-RUN PIP_NO_INPUT=1 pip3 install --no-cache-dir -e ".[CORE,OPTIONAL,DEV]"
-
-COPY --chown=fast_llm fast_llm/csrc/ ./fast_llm/csrc/
-RUN make -C ./fast_llm/csrc/
+# Copy the dependency files and install dependencies
+COPY --chown=fast_llm setup.py setup.cfg pyproject.toml ./
```

Comment:
Why adding the toml? It's not used for the installation.

Reply:
it appeared to me that the setup.* files and pyproject.toml fulfil similar purposes and could be grouped together. it is not uncommon to specify dependencies in the pyproject.toml file, because this is how setup tools usually works.

Reply:
We don't do that in fast-llm though, the toml file is just there because black needs it.

Reply:
ok, I'll split it up then and copy pyproject.toml somewhere else

Reply:
I don't really think it's worth copying at all... But if we do keep it let's keep it here so we don't add another line